src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -226,14 +226,18 @@ public DataAccessRequest createProgressReport(User user, DataAccessRequest progr
     validateProgressReport(user, progressReport, parentDar);
 
     String referenceId = progressReport.getReferenceId();
-    List<Integer> datasetIds = progressReport.getDatasetIds();
+    List<Integer> progressReportDatasetIds = progressReport.getDatasetIds();
+    List<Integer> darDatasetIds = dataAccessRequestDAO.findApprovedDatasetsByDar(parentDar.getReferenceId());
+    if(progressReportDatasetIds.stream().filter(datasetId -> !darDatasetIds.contains(datasetId)).toList().size() > 0) {
+      throw new BadRequestException("Progress report can only be created for approved datasets in the parent DAR");
+    }
```

Comment:
This is challenging for me to follow. I tried rewriting it a bit so I can wrap my brain around it:
```
    boolean validIds = new HashSet<>(darDatasetIds).containsAll(progressReportDatasetIds);
    if (!validIds) {
      throw new BadRequestException("Progress report can only be created for datasets in the parent DAR");
    }
```